src/artofillusion/polymesh/PolyMeshViewer.java

```diff
@@ -357,29 +357,17 @@ private void drawVertices()
             else
                 ref = i;
 
+            Color color = selected[ref] && visible[i] ? selectedVertColor : vertColor;
```

Comment:
`!(selected && visible)' is subtly different from `(!selected && visible)`.

When `(selected && !visible)`, or indeed `(!selected && !visible)` the vertice shouldn't be drawn at all. This can happen if the user hides a section of the model with `edit->Hide Selection`.